extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js

```diff
@@ -399,6 +399,11 @@ export function apply(ampdoc, element) {
 export function runCandidates(ampdoc, candidates) {
   return candidates.map(candidate =>
     whenLoaded(candidate).then(() => {
+      // <amp-img> will change the img's src inline data on unlayout and remove
+      // it from DOM, but a LOAD_END event would still be triggered afterwards.
+      if (candidate.signals().get(CommonSignals.UNLOAD)) {
```

Comment:
nit: should we move this check to `Criteria.meetsAll()` to make this method easier to read?

Reply:
Not sure it makes sense for it to be in that unit particularly since it would be have to be mocked in tests.

`whenLoaded` would make more sense, but I don't want to create superfluous wrapping promises for something that runs as many times as something as an `amp-img` appears on every page :(